### DISCUSSION TITLE
设置了文本换行后，如何给多行文本每一行行文本设置下划线

### DISCUSSION THREAD

**Discussion by Create-Peace** [antvis/S2#2951] [General]
设置了文本换行，多行文本，如何给每一行文本设置下划线

**Comment by lijinke666** [comment]
1. 自定义多行文本所在的单元格

```tsx
class CustomColCell extends ColCell {
  initCell() {
    super.initCell();

    // 绘制下划线
    const rects = this.getTextLineBoundingRects()
  }
}
```

2. `this.getTextLineBoundingRects()` 获取文本的包围盒信息
![image](https://github.com/user-attachments/assets/5d6cf677-2ec5-4cec-8c51-04c56430638f)

3. 根据文本的坐标绘制 Line 下划线, 使用 `renderLine` 或者 G 的 API绘制

```ts
import { renderLine } from '@antv/s2'

rects.forEach((rect) => {
  renderLine(this, {
    // 传入 x1, y1 等配置, 将图形绘制在 Canvas 中
  })
})
```

参考 https://github.com/antvis/S2/blob/a4ae432d2ddff8a3b44fdfb7abd96d6a2369c81b/packages/s2-core/src/cell/base-cell.ts#L525-L533

**Comment by Create-Peace** [comment]
![image](https://github.com/user-attachments/assets/89084ca9-1b30-42a7-a239-12962d83e016)
为什么数据出来是这样的，每个rect都是一样的，明细表使用这种数据还能计算出线如何渲染，透视没有办法计算。
